course7/lec7_en.md

```diff
@@ -0,0 +1,346 @@
+  let _x = 0 // The result is of single-value type, which is consistent with the function definition
+}
+```
+
+# Variables
+
+- In Moonbit, we can define temporary variables within a code block using `let mut`.
+
+```moonbit
+let mut x = 1
+x = 10 // The assignment operation is a command.
+```
+
+- In Moonbit, the fields of a struct are immutable by default, and we also allow mutable fields, which need to be marked with `mut`.
+
+```moonbit
+struct Ref[T] { mut val : T }
+
+fn init {
+  let ref: Ref[Int] = { val: 1 } // ref itself is just a data binding
+  ref.val = 10 // We can modify the fields of the struct
+  println(ref.val.to_string()) // Output 10
+}
+```
+
+# Variables
+
+- We can consider a struct with mutable fields as a reference.
+
+![](../pics/ref.drawio.svg)
+
+# Aliases
+
+- Two identifiers pointing to the same mutable data structure can be considered aliases.
+
+```moonbit
+fn alter(a: Ref[Int], b: Ref[Int]) {
+  a.val = 10
+  b.val = 20
+}
+
+fn init {
+  let x: Ref[Int] = { val : 1 }
+  alter(x, x)
+  println(x.val.to_string()) // The value of x.val will be changed twice.
+}
+```
+
+# Aliases
+
+- Two identifiers pointing to the same mutable data structure can be considered aliases.
+
+![](../pics/alias.drawio.svg)
+
+- Mutable variables need to be handled carefully.
+
+# Loops
+
+- Using variables, we can define loops.
+
+```moonbit
+<Define variables and initial values>
+while <Check if the condition for continuing the loop is met>, <Iterate the variable> {
+  <Commands to be repeated>
+}
+```
+
+- For example, we can perform n output operations repeatedly.
+
+```moonbit
+let mut i = 0
+while i < 2, i = i + 1 {
+  println("Output") // Repeat the output twice.
+}
+```
+
+# Loops
+
+- When we enter the loop,
+  - We check if the condition for continuing the loop is met.
+  - Execute commands.
+  - Iterate the variable.
+  - Repeat the above process.
+- For example:
+
+```moonbit
+let mut i = 0 // <-- At this point, i is equal to 0
+while i < 2, i = i + 1 { // <-- Here, we check if i < 2 is true.
+  println("Output") // <-- 0 < 2, so continue execution, output the first time.
+} // <-- At this point, we execute i = i + 2.
+```
+
+# Loops
+
+- When we enter the loop,
+  - We check if the condition for continuing the loop is met.
+  - Execute commands.
+  - Iterate the variable.
+  - Repeat the above process.
+- For example:
+
+```moonbit
+// At this point, i is equal to 1
+while i < 2, i = i + 1 { // <-- Here, we check if i < 2 is true.
+  println("Output") // <-- 1 < 2, so continue execution, output the second time.
+} // <-- At this point, we execute i = i + 2.
+```
+
+# Loops
+
+- When we enter the loop,
+  - We check if the condition for continuing the loop is met.
+  - Execute commands.
+  - Iterate the variable.
+  - Repeat the above process.
+- For example:
+
+```moonbit
+// At this point, i is equal to 2
+while i < 2, i = i + 1 { // <-- Here, we check if i < 2 is true, which is false.
```

Comment:
Update the while loop here please.

Reply:
but why put a Fibonacci function as example here inside chinese version